### PR TITLE
Xdr malware fixes 2

### PR DESCRIPTION
**Pull request by ssokolovich**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [ Mirroring issue + true/false positive handling](https://jira-hq.paloaltonetworks.local/browse/XSUP-14977)
fixes: [ServiceNow input bug](https://jira-hq.paloaltonetworks.local/browse/CIAC-3675)
fixes: [Changing Polling Playbook For XDR get Script -> new command](https://jira-hq.paloaltonetworks.local/browse/CIAC-3676)

## Description
- Changed in a conditional task the context path from Questions -> Answers.
- Added validation for data collection output.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexXDR/ReleaseNotes/4_8_9.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+#### Mappers
+##### Cortex XDR - Outgoing Mapper
+Added support of the common incident fields.
+
+#### Playbooks
+##### Cortex XDR Malware - Investigation And Response
+Fixed conditional task routing the Incident Classification flow.
+##### Cortex XDR - False Positive Incident Handling
+Added path for no selected sha256 to be added as approved IOCs.
```
Comment (altmannyarden) [inline Packs/CortexXDR/ReleaseNotes/4_8_9.md:10]: I dont understand this sentence, try to re-phrase it

File: Packs/CortexXDR/ReleaseNotes/4_8_9.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+##### Cortex XDR - True Positive Incident Handling
+- Added incident closure comments.
+- Changed default FileSHA256 input representing 'File Paths' ( instead of 'File Path').
```
Comment (altmannyarden) [inline Packs/CortexXDR/ReleaseNotes/4_8_9.md:13]: Changed the default from ... to ... 
(to make it clearer)

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+#### Playbooks
+##### Malware Investigation & Response Incident Handler
+- Fixed Playbook Inputs.
+- Remove un-required inputs in sub-playbooks.
+
+#### Wizards
+##### Malware Use Case Adoption Wizard
+Edited Wizard notes for the XDR integration set up.
```
Comment (altmannyarden) [inline Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md:9]: Updated XDR integration set up notes.

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+#### Playbooks
+##### Malware Investigation & Response Incident Handler
+- Fixed Playbook Inputs.
+- Remove un-required inputs in sub-playbooks.
```
Comment (altmannyarden) [inline Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md:5]: Removed unnecessary inputs....

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+#### Playbooks
+##### Malware Investigation & Response Incident Handler
+- Fixed Playbook Inputs.
```
Comment (altmannyarden) [inline Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md:4]: which ones?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/playbook-MDE_-_True_Positive_Incident_Handling.yml [hunk 6]
```diff
@@ -414,6 +414,15 @@ tasks:
...
             value:
               simple: SNOW
+        - operator: isEqualString
+          left:
+            value:
+              simple: inputs.TicketingSystemToUse
+            iscontext: true
+          right:
+            value:
+              simple: ServiceNow
```
Comment (altmannyarden) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/playbook-MDE_-_True_Positive_Incident_Handling.yml:424]: need to change here as well no? to contains?

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md (hunk 3 above)
Comment (ssokolovich) [reply Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md:9]: Done.

File: Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml [hunk 7]
```diff
@@ -342,8 +336,6 @@ tasks:
         simple: ${inputs.MaliciousTagName}
       RetrieveFile:
         simple: ${inputs.RetrieveFile}
-      RunInvestigationActivities:
-        simple: ${inputs.RunInvestigationActivities}
       TicketProjectName:
```
[the remark this answers is not in the document]
Comment (ssokolovich) [reply Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml:339]: So are 2 answers :) 
1. Due to PM's request, we had to remove most of the main inputs and try to shrink that as much as possible.
Hence, most of them just were not even defined in the main playbook.
2. Also, due to PM's request, we needed to rephrase some of those inputs, and probably forgot to change that in some places.

So based on what I have seen - three should be now changed on the logic, and even provide the ability to tweak each vendor's playbook inputs without actually detaching something (as if defining a sub playbook input to take a non-existed context key - will ignore the inputs' default values).

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md (hunk 4 above)
Comment (ssokolovich) [reply Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md:5]: Done

File: Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md (hunk 5 above)
Comment (ssokolovich) [reply Packs/MalwareInvestigationAndResponse/ReleaseNotes/1_5_8.md]:
```
##### Malware Investigation & Response Incident Handler
- Fixed Playbook Inputs 'SIEMEDRProductToUse' usage.
```

Is this better ?

**Comment by ShirleyDenkberg** [comment]
@altmannyarden Doc review completed.

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml [hunk 8]
```diff
@@ -57,7 +55,7 @@ inputs:
   playbookInputQuery:
   required: false
   value:
-    simple: ${incident.processpath}
+    simple: ${incident.processpaths}
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml:58]: if we can get multiple values here, lets add unique tranformer no?

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml [hunk 9]
```diff
@@ -963,7 +977,7 @@ tasks:
           simple: Confirm which File Path will be deleted
         options: []
         optionsarg:
-        - simple: ${incident.processpaths}
+        - simple: ${inputs.FilePaths}
```
Comment (altmannyarden) [inline Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml:980]: Same as the comment above (unique transformer)

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_8_12.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+##### CrowdStrike Falcon - True Positive Incident Handling
+Fixed an issue with the 'ServiceNow' ticketing system.
```
Comment (altmannyarden) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_8_12.md:4]: If you fixed an issue with ServiceNow ticketing system, the fix should be in the integration pack.
Please re phrase this :)

File: Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (altmannyarden) [reply Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml:339]: so its ok to remove those 4 inputs?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_11_8.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+##### MDE - True Positive Incident Handling
+Fixed an issue with the 'ServiceNow' ticketing system.
```
Comment (altmannyarden) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_11_8.md:4]: should re phrase here as well

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_8_12.md (hunk 10 above)
Comment (ssokolovich) [reply Packs/CrowdStrikeFalcon/ReleaseNotes/1_8_12.md]: Done.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_11_8.md (hunk 11 above)
Comment (ssokolovich) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_11_8.md]: Done.

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml (hunk 9 above)
Comment (ssokolovich) [reply Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml:989]: Done.

File: Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml (hunk 8 above)
Comment (ssokolovich) [reply Packs/CortexXDR/Playbooks/playbook-Cortex_XDR_-_True_Positive_Incident_Handling.yml]: Done.

File: Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (ssokolovich) [reply Packs/MalwareInvestigationAndResponse/Playbooks/playbook-Malware_Investigation_and_Response_Incident_Handler.yml:339]: Yes,
Those are the playbook arguments for the vendors' sub-playbooks.
The values for them are defined( most of them) in the sub-playbooks
Example:
image


The only one that It's not defined is the 'ActionTask' in MDE -> Which allows the user to decide if he wants to run any additional investigation actions ( AV scan, Collect Forensics package, Automated Investigation )
As this input, is not defined, the analyst should manually check which action he would like to do. as for now, this was overwritten by `null` (which equals to - 'do nothing')  through the master playbook.
So now, those that are didn't modify the playbooks will be able to use that option.

**Review by altmannyarden — APPROVED** [review]

**Pull request merged**